src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -331,11 +353,21 @@ public String buildPartitionPath(Record<GenericRecord> message,
 
         String encodePartition = partitioner.encodePartition(message, partitioningTimestamp);
         String partitionedPath = partitioner.generatePartitionedPath(message.getTopicName().get(), encodePartition);
+        String trimmedTopicPath = partitionedPath.replace(PATH_SEPARATOR + encodePartition, StringUtils.EMPTY);
```

Comment:
this is not good, undo things by default. This code runs many times. 
I'd suggest to move the login inside the partitioner class, then

Also maybe we should keep a local HashMap<String,String> with topic name -> path (without the date) so we don't recompute again and again the same pattern

Reply:
Also we need unit tests to check input / output with all the flags about the partitions and all the combinations